### ISSUE TITLE
GetUpdatesResponse often results to Null Point Exception 

### ISSUE THREAD

**Issue by Gummybearr** [pengrad/java-telegram-bot-api#234]
Hi I first want to say thank you for this great repo

        TelegramBot bot = new TelegramBot(Auth.Telegram.token);

        bot.setUpdatesListener(updates -> UpdatesListener.CONFIRMED_UPDATES_ALL);

        GetUpdates getUpdates = new GetUpdates().limit(10).offset(0);
        GetUpdatesResponse updatesResponse = bot.execute(getUpdates);

        for (Update update : updatesResponse.updates()) {
            System.out.println(update.updateId());
            System.out.println(update.message().text());
        }

This piece of code gets updated messages to the bot and then prints them if any exists.
However, this lines of codes sometimes work and sometimes results in Null Pointer Exception.
Yes I tested it in circumstances where there is no recent messages.

I guess it would be great if GetUpdatesResponse can be wrapped with optional in this case so that programmers can use this project in more safe environments:)

I might be wrong and it would be great if you can tell me if I am wrong with any points.
Also I will be more than happy to contribute to your project!

Have a great day:)

**Comment by torvicv** [comment]
I am using webhook and get same result.

**Comment by Sphirye** [comment]
Greetings, I have worked with this library for a while and I have noticed that this error occurs when you access a value within a property that does not exist.


You need to do a check that all the objects behind the target exist before to proccess them.

<h1>Example</h1>

In this case (Spaghetti Code alert, this sucks and probably exist another better way):

```
if (update.message() != null) {
   if (update.message().photo() != null) {
      if (update.message().caption() != null) {
         if (update.message().caption().equals("lol") {
            //Do something
         }
      }
   }
}
```

If using Kotlin:

```
if (update.message()!!.caption()!!.equals("lol") {
   //Do something
}
```

**Comment by Gummybearr** [comment]
Thanks for great code example Sphirye. But I'm not sure if your suggestion actually solves the problem. I might be wrong but I think the problem is that we sometimes receive null when we try to receive updates.

```kotlin
update.message()
```

This code accesses property(method) of variable "update". However, since "update" sometimes is null, no matter if we use polling or webhook, using update.message() still might result to NPE. I believe the argument whether it is appropriate to return null when the result is actually null has been resolved since people came up with the concept of Optional.

**Comment by pengrad** [comment]
Fixed in 5.2.0, listener will be called with empty list if updates is null
